system/datatypes/peer_reviewed_article.php

```diff
@@ -576,262 +239,235 @@ public function indexAction() {
 				$commentCountText = __('art-public-comment-count-plural', ['count' => $commentCount]);
 			}
 
-			$this->findBySelector('#pagename')->before('<div class="number_of_comments">'.$commentCountText.'</div>');
+			$this->html->find('#pagename')->before('<div class="number_of_comments">'.$commentCountText.'</div>');
 		}
 
 		if ($author) {
 			$main->append('<div class="author">' . __('art-by') . ' ' . htmlspecialchars($author) . '</div>');
 		}
 
-		$article = $content->getChild(self::FIELD_NAME_TEXT)->getHtml();
+		$article = $content->find(self::FIELD_NAME_TEXT)->getHtml();
 		$main->append('<div class="article">'.$article.'</div>');
 
-		$method = $context->getChild(self::FIELD_NAME_METHOD)->getHtml();
+		$method = $context->find(self::FIELD_NAME_METHOD)->getHtml();
 		if ($method) {
 			/** @var HyphaDomElement $methodContainer */
-			$methodContainer = $this->xml->createElement('div');
+			$methodContainer = $this->html->createElement('div');
 			$methodContainer->attr('class', 'method');
 			$methodContainer->append('<h2>' . __('art-method') . '</h2>');
 			$methodContainer->append($method);
 			$main->append($methodContainer);
 		}
 
-		$sources = $this->getDocPageByName(self::FIELD_NAME_SOURCES)->getHtml();
+		$sources = $this->xml->find(self::FIELD_NAME_SOURCES)->getHtml();
 		if (!empty($sources)) {
 			$main->append('<div class="sources"><h2>' . __('art-sources') . '</h2>' . $sources . '</div>');
 		}
 
 		if (isUser() && self::STATUS_DRAFT !== $status) {
-			/** @var HyphaDomElement $discussionsContainer */
-			$discussionsContainer = $this->xml->createElement('div');
-			$discussionsContainer->attr('class', 'review-comments-wrapper');
-			$discussionsContainer->append('<h2>' . __('art-review-comments') . '</h2>');
-			$this->updateDiscussionsContainer('review', $discussionsContainer);
-			$main->append($discussionsContainer);
-			$main->append($this->getApprovesContainer());
+			/** @var HyphaDomElement $discussionsDomElement */
+			$discussionsDomElement = $this->html->createElement('div');
+			$discussionsDomElement->attr('class', 'review-comments-wrapper');
+			$discussionsDomElement->append('<h2>' . __('art-review-comments') . '</h2>');
+			$this->appendChildrenToDiscussionsDomElement('review', $discussionsDomElement);
```

Comment:
I was a bit confused by the `appendChildrenToDiscussionsDomElement` method, I somehow expected it to add `$discussionsDomElement` somewhere into the HTML DOM (though I now see that happens in the next line). In reality, the method only adds existing comments to the container. Perhaps it should be named differently? `addDiscussionsToElement`, or `fillDiscussionsContainer` or something like that?

Reply:
changed name to `fillDiscussionsContainer`